internal/gh/lister_test.go

```diff
@@ -15,220 +15,297 @@
-						Title:  github.String("Issue 2"),
-						State:  github.String("open"),
-						Number: github.Int(456),
-					},
-				),
-			)
-			iss, err := GetIssue(456, WithClient(mockedHTTPClient),
-				WithProject("fakeorg/fakeproject"))
-			Expect(iss).NotTo(BeNil())
-			Expect(iss.GetNumber()).To(Equal(456))
-			Expect(err).NotTo(HaveOccurred())
+				)},
+			want:     []*github.Issue{},
+			wantErr:  true,
+			errMatch: "github went belly up or something",
+		},
+
+		// var _ = Describe("Lister", func() {
+		// 	Describe("ListIssues", func() {
+		// 		It("should return error if list fails", func() {
+		// 			// if our request returns an error ListIssues should return
+		// 			// that error
+		// 			mockedHTTPClient := mock.NewMockedHTTPClient(
+		// 				mock.WithRequestMatchHandler(
+		// 					mock.GetReposIssuesByOwnerByRepo,
+		// 					http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// 						mock.WriteError(
+		// 							w,
+		// 							http.StatusInternalServerError,
+		// 							"github went belly up or something",
+		// 						)
+		// 					}),
+		// 				),
+		// 			)
+		// 			iss, err := ListIssues(WithClient(mockedHTTPClient), WithProject("fakeorg/fakeproject"))
+		// 			Expect(iss).To(BeNil())
+		// 			Expect(err).To(HaveOccurred())
+		// 		})
+		// 		It("should return error if Options return an error", func() {
+		// 			_, err := ListIssues(func(c *ListerConfig) error {
+		// 				return fmt.Errorf("do you see me")
+		// 			})
+		// 			Expect(err).To(HaveOccurred())
+		// 			Expect(err.Error()).To(Equal("do you see me"))
+		// 		})
+		// 	})
+		// 	Describe("GetIssue", func() {
+		// 		It("should find the open issue", func() {
+		// 			mockedHTTPClient := mock.NewMockedHTTPClient(
+		// 				mock.WithRequestMatch(mock.GetReposIssuesByOwnerByRepoByIssueNumber,
+		// 					github.Issue{
+		// 						ID:     github.Int64(456),
+		// 						Title:  github.String("Issue 2"),
+		// 						State:  github.String("open"),
+		// 						Number: github.Int(456),
+		// 					},
+		// 				),
+		// 			)
+		// 			iss, err := GetIssue(456, WithClient(mockedHTTPClient),
+		// 				WithProject("fakeorg/fakeproject"))
+		// 			Expect(iss).NotTo(BeNil())
+		// 			Expect(iss.GetNumber()).To(Equal(456))
+		// 			Expect(err).NotTo(HaveOccurred())
+		// 		})
+		// 	})
+		// })
+	}
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			c, err := NewConnection(s.connectionOptions...)
+			require.NoError(t, err)
+
+			err = c.Connect()
+			require.NoError(t, err)
+
+			iss, err := c.ListIssues(s.options...)
+			if err != nil && s.wantErr != true {
+				t.Errorf("ListIssues() error = %v, wantErr %v", err, s.wantErr)
+				return
+			}
+			if err != nil && s.wantErr == true {
+				t.Errorf("ListIssues() error = %q, wantErr %v, errMatch %q", err, s.wantErr, s.errMatch)
+				require.ErrorContains(t, err, s.errMatch)
+			}
+			if !reflect.DeepEqual(iss, s.want) {
+				t.Errorf("ListIssues() = %v, want %v", iss, s.want)
+			}
```

Comment:
It was actually because the github.ErrorResponse fulfilled by the mock lib doesn't have a valid http.Response (nil) so any error.Error() call immediately segv'd.